descriptor/qsarmolecular/src/main/java/org/openscience/cdk/qsar/descriptors/molecular/ALOGPDescriptor.java

```diff
@@ -1340,19 +1367,19 @@ private void calcGroup066_to_079(int i) {
             boolean flag1 = false;
             boolean flag2 = false;
 
-            for (int j = 0; j <= ca.size() - 1; j++) {
-                if (((IAtom) ca.get(j)).getSymbol().equals("H")) continue;
-                if (atomContainer.getBond(ai, ((IAtom) ca.get(j))).getOrder() == IBond.Order.DOUBLE) {
-                    if (((IAtom) ca.get(j)).getSymbol().equals("C")) {
+            for (int j = 0; j <= nbors.size() - 1; j++) {
+                if (((IAtom) nbors.get(j)).getSymbol().equals("H")) continue;
+                if (atomContainer.getBond(ai, ((IAtom) nbors.get(j))).getOrder() == IBond.Order.DOUBLE) {
+                    if (((IAtom) nbors.get(j)).getSymbol().equals("C")) {
                         frags[74]++;
                         alogpfrag[i] = 74;
                         return;
                     } else {
                         flag1 = true;
                     }
                 } else {
-                    if (!((IAtom) ca.get(j)).getSymbol().equals("C")
-                            || ((IAtom) ca.get(j)).getFlag(CDKConstants.ISAROMATIC)) {
+                    if (!((IAtom) nbors.get(j)).getSymbol().equals("C")
```

Comment:
Can this casting to `IAtom` not be removed in all these lines too?

Reply:
Yes, but this class is 2000 lines long... and very likely to be rewritten I changed as little as possible to keep me sane :-).